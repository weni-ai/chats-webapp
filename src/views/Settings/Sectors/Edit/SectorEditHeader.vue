<template>
  <header class="sector-edit-view__header">
    <h1 class="header__title">
      {{ `${$t('sector.title')} ${sectorName}` }}
    </h1>
    <p class="header__description">
      {{ $t('config_chats.edit_sector.description') }}
    </p>
  </header>
</template>

<script>
export default {
  name: 'SectorEditHeader',

  props: {
    sectorName: {
      type: String,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.sector-edit-view__header {
  display: grid;
  align-items: center;
  gap: $unnnic-spacing-nano;

  .header__title {
    color: $unnnic-color-neutral-black;
    font-family: $unnnic-font-family-primary;
    font-size: $unnnic-font-size-title-sm;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-title-sm + $unnnic-line-height-md;
  }

  .header__description {
    color: $unnnic-color-neutral-dark;
    font-size: $unnnic-font-size-body-gt;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }
}
</style>
