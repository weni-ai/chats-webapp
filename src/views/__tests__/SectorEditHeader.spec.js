import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';

import SectorEditHeader from '@/views/Settings/Sectors/Edit/SectorEditHeader.vue';

const createWrapper = (props) => {
  return mount(SectorEditHeader, {
    props,
  });
};

describe('SectorEditHeader.vue', () => {
  let wrapper;
  const sectorName = 'Sector 1';

  beforeEach(() => {
    wrapper = createWrapper({ sectorName });
  });

  it('should renders the sector title and description', () => {
    const headerTitle = wrapper.find('[data-testid=header-title]');
    const headerDescription = wrapper.find('[data-testid=header-description]');

    expect(headerTitle.text()).toBe(
      `${wrapper.vm.$t('sector.title')} ${sectorName}`,
    );
    expect(headerDescription.text()).toBe(
      wrapper.vm.$t('config_chats.edit_sector.description'),
    );
  });

  it('Should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
