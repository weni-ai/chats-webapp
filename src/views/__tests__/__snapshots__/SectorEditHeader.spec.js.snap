// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SectorEditHeader.vue > Should match the snapshot 1`] = `
<header
  class="sector-edit-view__header"
  data-v-573bd910=""
>
  <h1
    class="header__title"
    data-testid="header-title"
    data-v-573bd910=""
  >
    Sector Sector 1
  </h1>
  <p
    class="header__description"
    data-testid="header-description"
    data-v-573bd910=""
  >
    config_chats.edit_sector.description
  </p>
</header>
`;
